nautobot/core/management/commands/generate_performance_test_endpoints.py

```diff
@@ -0,0 +1,269 @@
+        """
+        if not model_class:
+            # A generic endpoint like `core:home`
+            if view_name not in self.app_name_to_urls["endpoints"]:
+                self.app_name_to_urls["endpoints"][view_name] = []
+            self.app_name_to_urls["endpoints"][view_name].append(url_pattern)
+            return
+
+        # If the model class is found, then we know we are dealing with a model related endpoint
+        if len(model_class.objects.all()) < 2:
+            # TODO handle the case where there is no instances of the model is found
+            self.stderr.write(f"Not enough instances of {model_class} found, need at least 2")
+            return
+
+        # Handle detail view url patterns
+        if "_list" not in view_name and "-list" not in view_name:
+            # Identify the placeholder for the uuid
+            replace_string = ""
+            if "<uuid:pk>" in url_pattern:
+                replace_string = "<uuid:pk>"
+            elif "(?P<pk>[/.]+)" in url_pattern:
+                replace_string = "(?P<pk>[/.]+)"
+
+            if replace_string:
+                # Replace the uuid with the actual uuid
+                first_url_pattern = url_pattern.replace(replace_string, str(model_class.objects.first().pk))
+                second_url_pattern = url_pattern.replace(replace_string, str(model_class.objects.last().pk))
+                if view_name not in self.app_name_to_urls["endpoints"]:
+                    self.app_name_to_urls["endpoints"][view_name] = []
+                self.app_name_to_urls["endpoints"][view_name].append(first_url_pattern)
+                self.app_name_to_urls["endpoints"][view_name].append(second_url_pattern)
+        # Handle list view url patterns
+        else:
+            if view_name not in self.app_name_to_urls["endpoints"]:
+                self.app_name_to_urls["endpoints"][view_name] = []
+            # One endpoint with default pagination
+            self.app_name_to_urls["endpoints"][view_name].append(url_pattern)
+            total_count = len(model_class.objects.all())
+            page_query_parameter = 5
+            per_page_query_parameter = total_count // page_query_parameter
+            if not is_api_endpoint:
+                query_params = urlencode(
+                    {
+                        "per_page": per_page_query_parameter,
+                        "page": page_query_parameter,
+                    }
+                )
+            else:
+                query_params = urlencode(
+                    {
+                        "limit": per_page_query_parameter,
+                        "offset": per_page_query_parameter * (page_query_parameter - 1),
+                    }
+                )
+            # One endpoint with default pagination
+            self.app_name_to_urls["endpoints"][view_name].append(url_pattern + f"?{query_params}")
+
+    def construct_view_name_and_url_pattern(self, pattern) -> tuple[str, str, bool]:
```

Comment:
Since the returned `url_pattern` and `view_name` can be `None`:

```suggestion
    def construct_view_name_and_url_pattern(self, pattern) -> tuple[Optional[str], Optional[str], bool]:
```